### ISSUE TITLE
deployment is successful but ceph osd tree shows no osds!

### ISSUE THREAD

**Issue by Masber** [ceph/ceph-ansible#1782]
Hi,
deployment is successful but `ceph osd tree` shows no osds.

```
[root@kolla ceph-ansible]# ssh ceph-4-2
Last login: Tue Aug 22 18:16:15 2017 from 10.1.0.10
[root@ceph-4-2 ~]# ceph osd tree
ID WEIGHT TYPE NAME    UP/DOWN REWEIGHT PRIMARY-AFFINITY
-1      0 root default
[root@ceph-4-2 ~]# ceph -s
    cluster 1439d592-555b-4ebc-89f7-4d93df2ae4b0
     health HEALTH_ERR
            72 pgs are stuck inactive for more than 300 seconds
            72 pgs stuck inactive
            72 pgs stuck unclean
            no osds
     monmap e2: 1 mons at {ceph-4-2=10.1.0.42:6789/0}
            election epoch 5, quorum 0 ceph-4-2
        mgr no daemons active
     osdmap e2: 0 osds: 0 up, 0 in
            flags sortbitwise,require_jewel_osds,require_kraken_osds
      pgmap v3: 72 pgs, 2 pools, 0 bytes data, 0 objects
            0 kB used, 0 kB / 0 kB avail
                  72 creating
```

it is weird, looks like it didn't formatted the disks properly:

```
[root@ceph-4-2 ~]# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda           8:0    0 59.6G  0 disk
├─sda1        8:1    0    1G  0 part /boot
├─sda2        8:2    0    4G  0 part [SWAP]
└─sda3        8:3    0 54.6G  0 part /
nvme0n1     259:21   0  1.8T  0 disk
├─nvme0n1p1 259:22   0  1.8T  0 part /var/lib/ceph/osd/ceph-0
└─nvme0n1p2 259:23   0    5G  0 part
nvme1n1     259:6    0  1.8T  0 disk
├─nvme1n1p1 259:7    0  1.8T  0 part /var/lib/ceph/osd/ceph-1
└─nvme1n1p2 259:8    0    5G  0 part
nvme2n1     259:0    0  1.8T  0 disk
├─nvme2n1p1 259:1    0  1.8T  0 part /var/lib/ceph/osd/ceph-2
└─nvme2n1p2 259:2    0    5G  0 part
nvme3n1     259:24   0  1.8T  0 disk
├─nvme3n1p1 259:25   0  1.8T  0 part
└─nvme3n1p2 259:26   0    5G  0 part
nvme4n1     259:9    0  1.8T  0 disk
├─nvme4n1p1 259:10   0  1.8T  0 part /var/lib/ceph/osd/ceph-4
└─nvme4n1p2 259:11   0    5G  0 part
nvme5n1     259:3    0  1.8T  0 disk
├─nvme5n1p1 259:4    0  1.8T  0 part /var/lib/ceph/osd/ceph-5
└─nvme5n1p2 259:5    0    5G  0 part
nvme6n1     259:15   0  1.8T  0 disk
├─nvme6n1p1 259:16   0  1.8T  0 part /var/lib/ceph/osd/ceph-6
└─nvme6n1p2 259:17   0    5G  0 part
nvme7n1     259:27   0  1.8T  0 disk
├─nvme7n1p1 259:28   0  1.8T  0 part /var/lib/ceph/osd/ceph-7
└─nvme7n1p2 259:29   0    5G  0 part
nvme8n1     259:12   0  1.8T  0 disk
├─nvme8n1p1 259:13   0  1.8T  0 part /var/lib/ceph/osd/ceph-8
└─nvme8n1p2 259:14   0    5G  0 part
nvme9n1     259:18   0  1.8T  0 disk
├─nvme9n1p1 259:19   0  1.8T  0 part /var/lib/ceph/osd/ceph-9
└─nvme9n1p2 259:20   0    5G  0 part

```
[playbook.zip](https://github.com/ceph/ceph-ansible/files/1241280/playbook.zip)

[group_vars and hosts.zip](https://github.com/ceph/ceph-ansible/files/1241287/group_vars.and.hosts.zip)

Any advise would be much appreciated

thank you very much

**Comment by mistur** [comment]
I'm not absolutly sure, but I think you should put disk device into osds.yml and not partition, ceph-disk will create the partition 

```devices:
  - /dev/nvme0
  - /dev/nvme1
  - /dev/nvme2
  - /dev/nvme3
  - /dev/nvme4
  - /dev/nvme5
  - /dev/nvme6
  - /dev/nvme7
  - /dev/nvme8
  - /dev/nvme9```

Yoann

**Comment by Masber** [comment]
@mistur 
I changed to:
  - nvme0n1
  - nvme1n1
  - nvme2n1
  - nvme3n1
  - nvme4n1
  - nvme5n1
  - nvme6n1
  - nvme7n1
  - nvme8n1
  - nvme9n1

 but got an error message:

```
TASK [ceph-osd : fix partitions gpt header or labels of the osd disks] *********
failed: [ceph-4-3] (item=[{u'_ansible_parsed': True, u'cmd': u'parted --script nvme0n1 print > /dev/null 2>&1', u'end': u'2017-08-22 18:38:19.468588', u'_ansible_no_log': False, u'stdout': u'', u'_ansible_item_result': True, u'changed': False, u'item': [{u'_ansible_parsed': True, u'_ansible_item_result': True, u'end': u'2017-08-22 18:38:16.175208', u'_ansible_no_log': False, u'stdout': u'', u'cmd': u"readlink -f nvme0n1 | egrep '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}|fio[a-z]{1,2}[0-9]{1,2}$'", u'rc': 1, u'item': u'nvme0n1', u'delta': u'0:00:00.007995', u'stderr': u'', u'changed': False, u'invocation': {u'module_name': u'command', u'module_args': {u'warn': True, u'executable': None, u'_uses_shell': True, u'_raw_params': u"readlink -f nvme0n1 | egrep '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}|fio[a-z]{1,2}[0-9]{1,2}$'", u'removes': None, u'creates': None, u'chdir': None}}, u'stdout_lines': [], u'failed_when_result': False, u'start': u'2017-08-22 18:38:16.167213', u'warnings': [], u'failed': False}, u'nvme0n1'], u'delta': u'0:00:00.009862', u'stderr': u'', u'rc': 1, u'invocation': {u'module_name': u'command', u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'_raw_params': u'parted --script nvme0n1 print > /dev/null 2>&1', u'removes': None, u'warn': True, u'chdir': None}}, u'stdout_lines': [], u'failed_when_result': False, u'start': u'2017-08-22 18:38:19.458726', u'warnings': [], u'failed': False}, u'nvme0n1']) => {"changed": false, "cmd": "sgdisk --zap-all --clear --mbrtogpt -- nvme0n1 || sgdisk --zap-all --clear --mbrtogpt -- nvme0n1", "delta": "0:00:00.028089", "end": "2017-08-22 18:38:22.875024", "failed": true, "item": [{"_ansible_item_result": true, "_ansible_no_log": false, "_ansible_parsed": true, "changed": false, "cmd": "parted --script nvme0n1 print > /dev/null 2>&1", "delta": "0:00:00.009862", "end": "2017-08-22 18:38:19.468588", "failed": false, "failed_when_result": false, "invocation": {"module_args": {"_raw_params": "parted --script nvme0n1 print > /dev/null 2>&1", "_uses_shell": true, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "item": [{"_ansible_item_result": true, "_ansible_no_log": false, "_ansible_parsed": true, "changed": false, "cmd": "readlink -f nvme0n1 | egrep '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}|fio[a-z]{1,2}[0-9]{1,2}$'", "delta": "0:00:00.007995", "end": "2017-08-22 18:38:16.175208", "failed": false, "failed_when_result": false, "invocation": {"module_args": {"_raw_params": "readlink -f nvme0n1 | egrep '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}|fio[a-z]{1,2}[0-9]{1,2}$'", "_uses_shell": true, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "item": "nvme0n1", "rc": 1, "start": "2017-08-22 18:38:16.167213", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "nvme0n1"], "rc": 1, "start": "2017-08-22 18:38:19.458726", "stderr": "", "stdout": "", "stdout_lines": [], "warnings": []}, "nvme0n1"], "rc": 3, "start": "2017-08-22 18:38:22.846935", "stderr": "Problem opening nvme0n1 for reading! Error is 2.\nThe specified file does not exist!\nProblem opening '' for writing! Program will now terminate.\nWarning! MBR not overwritten! Error is 2!\nProblem opening nvme0n1 for reading! Error is 2.\nThe specified file does not exist!\nProblem opening '' for writing! Program will now terminate.\nWarning! MBR not overwritten! Error is 2!", "stdout": "Information: Creating fresh partition table; will override earlier problems!\nNon-GPT disk; not saving changes. Use -g to override.\nInformation: Creating fresh partition table; will override earlier problems!\nNon-GPT disk; not saving changes. Use -g to override.", "stdout_lines": ["Information: Creating fresh partition table; will override earlier problems!", "Non-GPT disk; not saving changes. Use -g to override.", "Information: Creating fresh partition table; will override earlier problems!", "Non-GPT disk; not saving changes. Use -g to override."], "warnings": []}
```

**Comment by mistur** [comment]
ok, I thought n1 was the partition number, I was wrong. You need the full path /dev/nvmeXn1, your previous osds.yml was OK

**Comment by mistur** [comment]
Could you revert the change and try to run the playbook again ?
```
  - /dev/nvme0n1
  - /dev/nvme1n1
  - /dev/nvme2n1
  - /dev/nvme3n1
  - /dev/nvme4n1
  - /dev/nvme5n1
  - /dev/nvme6n1
  - /dev/nvme7n1
  - /dev/nvme8n1
  - /dev/nvme9n1
```

**Comment by Masber** [comment]
I got same result
[playbook.txt](https://github.com/ceph/ceph-ansible/files/1243809/playbook.txt)

Not sure why disk nvme3n1 is not mounted?

```
[root@kolla ceph-ansible]# ssh ceph-4-2
Last login: Wed Aug 23 11:03:37 2017 from 10.1.0.10
[root@ceph-4-2 ~]# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda           8:0    0 59.6G  0 disk
├─sda1        8:1    0    1G  0 part /boot
├─sda2        8:2    0    4G  0 part [SWAP]
└─sda3        8:3    0 54.6G  0 part /
nvme0n1     259:13   0  1.8T  0 disk
├─nvme0n1p1 259:16   0  1.8T  0 part /var/lib/ceph/osd/ceph-0
└─nvme0n1p2 259:17   0    5G  0 part
nvme1n1     259:3    0  1.8T  0 disk
├─nvme1n1p1 259:5    0  1.8T  0 part /var/lib/ceph/osd/ceph-1
└─nvme1n1p2 259:6    0    5G  0 part
nvme2n1     259:9    0  1.8T  0 disk
├─nvme2n1p1 259:10   0  1.8T  0 part /var/lib/ceph/osd/ceph-2
└─nvme2n1p2 259:11   0    5G  0 part
nvme3n1     259:18   0  1.8T  0 disk
├─nvme3n1p1 259:19   0  1.8T  0 part
└─nvme3n1p2 259:20   0    5G  0 part
nvme4n1     259:12   0  1.8T  0 disk
├─nvme4n1p1 259:14   0  1.8T  0 part /var/lib/ceph/osd/ceph-4
└─nvme4n1p2 259:15   0    5G  0 part
nvme5n1     259:27   0  1.8T  0 disk
├─nvme5n1p1 259:28   0  1.8T  0 part /var/lib/ceph/osd/ceph-5
└─nvme5n1p2 259:29   0    5G  0 part
nvme6n1     259:0    0  1.8T  0 disk
├─nvme6n1p1 259:1    0  1.8T  0 part /var/lib/ceph/osd/ceph-6
└─nvme6n1p2 259:2    0    5G  0 part
nvme7n1     259:21   0  1.8T  0 disk
├─nvme7n1p1 259:22   0  1.8T  0 part /var/lib/ceph/osd/ceph-7
└─nvme7n1p2 259:23   0    5G  0 part
nvme8n1     259:4    0  1.8T  0 disk
├─nvme8n1p1 259:7    0  1.8T  0 part /var/lib/ceph/osd/ceph-8
└─nvme8n1p2 259:8    0    5G  0 part
nvme9n1     259:24   0  1.8T  0 disk
├─nvme9n1p1 259:25   0  1.8T  0 part /var/lib/ceph/osd/ceph-9
└─nvme9n1p2 259:26   0    5G  0 part

```

**Comment by mistur** [comment]
Hello,

at the end of the log you can see 

```
TASK [ceph-osd : ensure osd daemons are started] *******************************
failed: [ceph-4-2] (item=0) => {"changed": false, "failed": true, "item": "0", "msg": "Unable to start service ceph-osd@0: Job for ceph-osd@0.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@0.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@0.service\" followed by \"systemctl start ceph-osd@0.service\" again.\n"}
failed: [ceph-4-2] (item=1) => {"changed": false, "failed": true, "item": "1", "msg": "Unable to start service ceph-osd@1: Job for ceph-osd@1.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@1.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@1.service\" followed by \"systemctl start ceph-osd@1.service\" again.\n"}
failed: [ceph-4-2] (item=2) => {"changed": false, "failed": true, "item": "2", "msg": "Unable to start service ceph-osd@2: Job for ceph-osd@2.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@2.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@2.service\" followed by \"systemctl start ceph-osd@2.service\" again.\n"}
failed: [ceph-4-2] (item=4) => {"changed": false, "failed": true, "item": "4", "msg": "Unable to start service ceph-osd@4: Job for ceph-osd@4.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@4.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@4.service\" followed by \"systemctl start ceph-osd@4.service\" again.\n"}
failed: [ceph-4-2] (item=5) => {"changed": false, "failed": true, "item": "5", "msg": "Unable to start service ceph-osd@5: Job for ceph-osd@5.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@5.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@5.service\" followed by \"systemctl start ceph-osd@5.service\" again.\n"}
failed: [ceph-4-2] (item=6) => {"changed": false, "failed": true, "item": "6", "msg": "Unable to start service ceph-osd@6: Job for ceph-osd@6.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@6.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@6.service\" followed by \"systemctl start ceph-osd@6.service\" again.\n"}
failed: [ceph-4-2] (item=7) => {"changed": false, "failed": true, "item": "7", "msg": "Unable to start service ceph-osd@7: Job for ceph-osd@7.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@7.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@7.service\" followed by \"systemctl start ceph-osd@7.service\" again.\n"}
failed: [ceph-4-2] (item=8) => {"changed": false, "failed": true, "item": "8", "msg": "Unable to start service ceph-osd@8: Job for ceph-osd@8.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@8.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@8.service\" followed by \"systemctl start ceph-osd@8.service\" again.\n"}
failed: [ceph-4-2] (item=9) => {"changed": false, "failed": true, "item": "9", "msg": "Unable to start service ceph-osd@9: Job for ceph-osd@9.service failed because start of the service was attempted too often. See \"systemctl status ceph-osd@9.service\" and \"journalctl -xe\" for details.\nTo force a start use \"systemctl reset-failed ceph-osd@9.service\" followed by \"systemctl start ceph-osd@9.service\" again.\n"}
```

it seem to have an issue with the state of services on systemd, you might try to run systemctl reset-failed ceph-osd@X.service for each OSD then try to replay the playbook

I don't see any specific error for nvme3n1on ceph-4-2

Yoann

**Comment by leseb** [comment]
Got it, working on a patch.

**Comment by Masber** [comment]
this is what I am getting on release-2.2

```
TASK [ceph-osd : include] ***********************************************************************************************************************************************************************************************************************************
fatal: [ceph-4-2]: FAILED! => {"failed": true, "reason": "Syntax Error while loading YAML.\n\n\nThe error appears to have been in '/root/ceph-ansible/roles/ceph-osd/tasks/activate_osds.yml': line 18, column 61, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n- name: activate osd(s) when device is a disk\n  command: \"ceph-disk activate {{ item.1 | regex_replace('^(\\/dev\\/cciss\\/c[0-9]{1}d[0-9]{1})$', '\\\\1p') }}1\"\n                                                            ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
fatal: [ceph-4-3]: FAILED! => {"failed": true, "reason": "Syntax Error while loading YAML.\n\n\nThe error appears to have been in '/root/ceph-ansible/roles/ceph-osd/tasks/activate_osds.yml': line 18, column 61, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n- name: activate osd(s) when device is a disk\n  command: \"ceph-disk activate {{ item.1 | regex_replace('^(\\/dev\\/cciss\\/c[0-9]{1}d[0-9]{1})$', '\\\\1p') }}1\"\n                                                            ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
```

**Comment by leseb** [comment]
@Masber I need you to test #1803 not release-2.2. Thanks.

**Comment by Masber** [comment]
@leseb 

ceph osd tree only shows 1 node (should be 2)

```
[root@ceph-4-2 ~]# ceph osd tree
ID WEIGHT   TYPE NAME         UP/DOWN REWEIGHT PRIMARY-AFFINITY
-1 18.08685 root default
-2 18.08685     host ceph-4-2
 0  1.80869         osd.0          up  1.00000          1.00000
 1  1.80869         osd.1          up  1.00000          1.00000
 2  1.80869         osd.2          up  1.00000          1.00000
 3  1.80869         osd.3          up  1.00000          1.00000
 4  1.80869         osd.4          up  1.00000          1.00000
 5  1.80869         osd.5          up  1.00000          1.00000
 6  1.80869         osd.6          up  1.00000          1.00000
 7  1.80869         osd.7          up  1.00000          1.00000
 8  1.80869         osd.8          up  1.00000          1.00000
 9  1.80869         osd.9          up  1.00000          1.00000
[root@ceph-4-2 ~]#
```
please find attached new log file

[playbook.txt](https://github.com/ceph/ceph-ansible/files/1248977/playbook.txt)

**Comment by leseb** [comment]
At least, it's a progress :).
I don't see much in the play, but it look like something went wrong.

Are the OSD partitionned on ceph-4-3?

**Comment by Masber** [comment]
@leseb thank you very much for this

yes, ceph-4-3 osd are partitioned but all partitions (nvme4n1p1) are mounted.
```
[root@ceph-4-3 ~]# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda           8:0    0 59.6G  0 disk
├─sda1        8:1    0    1G  0 part /boot
├─sda2        8:2    0    4G  0 part [SWAP]
└─sda3        8:3    0 54.6G  0 part /
sr0          11:0    1  680M  0 rom
nvme0n1     259:0    0  1.8T  0 disk
├─nvme0n1p1 259:10   0  1.8T  0 part /var/lib/ceph/tmp/mnt.sCq7zq
└─nvme0n1p2 259:11   0   10G  0 part
nvme1n1     259:7    0  1.8T  0 disk
├─nvme1n1p1 259:14   0  1.8T  0 part /var/lib/ceph/tmp/mnt.8ptcwO
└─nvme1n1p2 259:15   0   10G  0 part
nvme2n1     259:6    0  1.8T  0 disk
├─nvme2n1p1 259:12   0  1.8T  0 part /var/lib/ceph/tmp/mnt.oMwrjm
└─nvme2n1p2 259:13   0   10G  0 part
nvme3n1     259:2    0  1.8T  0 disk
├─nvme3n1p1 259:18   0  1.8T  0 part /var/lib/ceph/tmp/mnt.n8r6Es
└─nvme3n1p2 259:19   0   10G  0 part
nvme4n1     259:8    0  1.8T  0 disk
├─nvme4n1p1 259:20   0  1.8T  0 part
└─nvme4n1p2 259:21   0   10G  0 part
nvme5n1     259:3    0  1.8T  0 disk
├─nvme5n1p1 259:16   0  1.8T  0 part /var/lib/ceph/tmp/mnt.SNtaZn
└─nvme5n1p2 259:17   0   10G  0 part
nvme6n1     259:5    0  1.8T  0 disk
├─nvme6n1p1 259:22   0  1.8T  0 part /var/lib/ceph/tmp/mnt.FnNqKZ
└─nvme6n1p2 259:23   0   10G  0 part
nvme7n1     259:4    0  1.8T  0 disk
├─nvme7n1p1 259:26   0  1.8T  0 part /var/lib/ceph/tmp/mnt.Fz1hg1
└─nvme7n1p2 259:27   0   10G  0 part
nvme8n1     259:1    0  1.8T  0 disk
├─nvme8n1p1 259:28   0  1.8T  0 part /var/lib/ceph/tmp/mnt.lEfhTP
└─nvme8n1p2 259:29   0   10G  0 part
nvme9n1     259:9    0  1.8T  0 disk
├─nvme9n1p1 259:30   0  1.8T  0 part /var/lib/ceph/tmp/mnt.pf9OSF
└─nvme9n1p2 259:31   0   10G  0 part
```

Also ceph command times out
```
[root@ceph-4-3 ~]# ceph
2017-08-25 00:44:38.358572 7feeedc5c700  0 monclient(hunting): authenticate timed out after 300
2017-08-25 00:44:38.358619 7feeedc5c700  0 librados: client.admin authentication error (110) Connection timed out
Error connecting to cluster: TimedOut
```

**Comment by Masber** [comment]
ok I think I fixed my issue after opening the ports on local firewall

sudo firewall-cmd --zone=public --add-service=ceph-mon --permanent
sudo firewall-cmd --zone=public --add-service=ceph --permanent
sudo firewall-cmd --reload

**Comment by leseb** [comment]
@Masber glad to hear it's solved, I'll merge #1803 soon, still trying to figure out why the CI is failing.
